components/global/Nav/Nav.tsx

```diff
@@ -15,36 +17,89 @@ const isFirstBranchMatched = (pagePath: string, href: string) => {
   return href === sub
 }
 
-export const Nav: React.FC<NavProps> = ({ pagePath, menu }) => {
-  const [expanded, setExpanded] = React.useState(false)
-  const componentId = React.useRef(`nav-${uuid()}`)
-  const navRef = React.useRef<HTMLUListElement | undefined>(undefined)
+function isFocusableElement(element: Element): element is HTMLElement {
+  return 'focus' in element
+}
+
+export const Nav: React.FC<NavProps> = ({ menu }) => {
+  const dispatch = useNavigationDispatch()
+  const containerRef = React.useRef<HTMLDivElement>()
+  const navRef = React.useRef<HTMLUListElement>()
+  const activeElement = React.useRef<Element>()
+  const { pathname } = useRouter()
+  const { id, expanded } = useNavigationState()
+  const [status, setStatus] = React.useState<NavStatus>('closed')
+
+  React.useEffect(() => {
```

Comment:
Why do we need to use JavaScript to model active / focussed elements rather than relying on CSS? (genuine question)

Reply:
This is around accessibility. Because clicking that button causes the navigation to slide in we move the user's focus there, otherwise their focus would remain on the button and you wouldn't really know anything happened. Once the menu closes, we move focus back to where they started. There's no real way to do that with HTML/CSS alone.